torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -282,6 +282,61 @@ void setstateTuple(
     }
   }
 }
+
+// Check if the global static map of backend debug info
+// contains debug info for this module and any of its children.
+// If so combine all the maps together and return one.
+void getBackendDebugInfoMap(
+    const Module& m,
+    DelegateDebugInfoMapType& debug_map) {
+  const auto& map =
+      getStaticBackendModuleDebugInfoMapPtr()->getDebugInfoMap(m._ivalue());
+  if (map) {
+    debug_map.insert(map.value().begin(), map.value().end());
+  }
+  for (const auto& c : m.children()) {
+    getBackendDebugInfoMap(c, debug_map);
+  }
+}
+
+SourceRangeRecords getBackendSourceRanges(const Module& m) {
+  constexpr size_t kSourceRange = 1;
+  SourceRangeRecords sr_records;
+  const auto& map =
+      getStaticBackendModuleDebugInfoMapPtr()->getDebugInfoMap(m._ivalue());
+  if (map) {
+    const auto& map_val = map.value();
+    // This map is map of debug handle-to-delegateDebugInfoType
+    // DebugInfoPair = <source range, inlined_cs_ptr>
+    for (const auto& it : map_val) {
+      sr_records.emplace_back(
```

Comment:
why are all these set to max()?

Reply:
So these are `SourceRangeRecords` which are `std::vector<TaggedRanges>` where `TaggedRange = struct{byte_offset(uint64_t), source range}`. When module is not lowered to backend, the source range information contains byte offset. byte offset is the offset in the compiled code and JIT uses this find appropriate source range for debug. For the modules lowered to backend we will use debug handles to help us find source range and byte offsets wont be used. So thats why I have set it to max here to indicate they are invalid as it is an unlikely value for it.